### PR TITLE
Login+register

### PR DESCRIPTION
**Pull request by afifisuha**
build the login logout and register files

### CODE REVIEW

**Comment by DenisKent** [comment]
Noor not in so not able to review

File: client/src/pages/RegisterPage/RegisterPage.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    username:"a",
+    postCode:"1da",
+    houseNumber: 1,
+    phoneNo: "07787392473",
+    email:"foundersandeaters@gmail.com",
+    password:"1234",
+    passwordConfirm:"1234"
+  }
+})(ConnectedRegisterPage);
+
```
Comment (jema28) [inline client/src/pages/RegisterPage/RegisterPage.js]: LINT YOUR FILES please

File: client/src/pages/LoginPage/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    event.preventDefault();
+    this.props.userLogin({username,password});
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <h1>Login</h1>
+        <form onSubmit={this.handleSubmit}>
+          <Field component="input" type="text" name="username" />
```
Comment (shiryz) [inline client/src/pages/LoginPage/LoginPage.js]: Every field should have a label

File: client/src/pages/RegisterPage/RegisterPage.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    event.preventDefault();
+    this.props.userRegister({username, postCode, houseNo, phoneNo, email, password});
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <h1>Register</h1>
+        <form onSubmit={this.handleSubmit}>
+          <Field component="input" type="text" name="username" />
```
Comment (shiryz) [inline client/src/pages/RegisterPage/RegisterPage.js]: same here, add labels

File: client/src/pages/RegisterPage/RegisterPage.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          <Field component="input" type="text" name="postCode" />
+          <Field component="input" type="number" name="houseNo" />
+          <Field component="input" type="text" name="phoneNo" />
+          <Field component="input" type="email" name="email" />
+          <Field component="input" type="password" name="password" />
+          <Field component="input" type="password" name="passwordConfirm" />
+          <button>Sign up</button>
+          <Link to="/login">Login</Link>
```
Comment (shiryz) [inline client/src/pages/RegisterPage/RegisterPage.js]: This link is breaking your app 
`Uncaught TypeError: Cannot read property 'username' of undefined`

File: client/src/pages/RegisterPage/RegisterPage.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+const ConnectedRegisterPage = connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(RegisterPage);
+
+export default reduxForm({
+  form: "loginForm",
+  initialValues: {
```
Comment (shiryz) [inline client/src/pages/RegisterPage/RegisterPage.js]: Why is this here as an initial value?

File: server/controllers/userLogin.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+const { compareHash } = require('../utils/hashPassword');
+const { createJWT } = require('../utils/jwtHelper');
+
+exports.post = (req, res) => {
+  const { username, password } = req.body;
+  getHashedPass(username)
+    .then(hashedRow => compareHash(password, hashedRow))
+    .then(({ id: userId }) => {
+      res.cookie('jwt', createJWT(userId), {
+        maxAge: 1000 * 60 * 60 * 30,
```
Comment (shiryz) [inline server/controllers/userLogin.js]: Where did this number come from?

File: server/controllers/userRegister.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const { postCreateUser } = require('../database/postQueries/postCreateUser');
+const { hashPassword } = require('../utils/hashPassword');
+const { createJWT } = require('../utils/jwtHelper');
+
+exports.post = (req, res) => {
+  const { password } = req.body;
+  hashPassword(password)
+    .then(hash => postCreateUser({ ...req.body, password: hash }))
+    .then((userId) => {
+      res.cookie('jwt', createJWT(userId), { maxAge: 1000 * 60 * 60 * 30, httpOnly: true });
```
Comment (shiryz) [inline server/controllers/userRegister.js]: Why are you setting JWT in 2 files? you can register and redirect to login again so it would set in 1 place 
This is more of an enhancement you don't have to change it

File: server/controllers/userLogin.js (hunk 6 above)
Comment (DenisKent) [reply server/controllers/userLogin.js]: 30 days

File: client/src/pages/RegisterPage/RegisterPage.js (hunk 4 above)
Comment (DenisKent) [reply client/src/pages/RegisterPage/RegisterPage.js]: Struggled to reproduce the error. The initial values in the comment below were put in place to prevent errors like this.

File: client/src/pages/RegisterPage/RegisterPage.js (hunk 5 above)
Comment (DenisKent) [reply client/src/pages/RegisterPage/RegisterPage.js]: For the reason above

File: server/controllers/userLogin.js (hunk 6 above)
Comment (DenisKent) [reply server/controllers/userLogin.js]: Done

File: server/controllers/userRegister.js (hunk 7 above)
Comment (DenisKent) [reply server/controllers/userRegister.js]: Refactored the code so it is only written once.

File: client/src/pages/RegisterPage/RegisterPage.js (hunk 4 above)
Comment (shiryz) [reply client/src/pages/RegisterPage/RegisterPage.js]: the error happens when you go back to login from register page, currently, it is not working and I can't merge it in this state... I would suggest opening an issue if you don't want to work on fixing it at the moment
This might help https://stackoverflow.com/questions/30115324/pass-props-in-link-react-router

**Comment by DenisKent** [comment]
@jema28 can we style afterwards

**Comment by DenisKent** [comment]
@shiryz all ready I hope
